### PR TITLE
Update stb

### PR DESCRIPTION
**Pull request by mataha**
This PR attempts to update `stb` in the most straightforward way possible as well as include fixes from main repo's unmerged PRs for cases rearing their ugly heads during everyday usage:

 - stb#1299: stb_rect_pack: Make rect_height_compare a stable sort
 - stb#1402: stb_image: Fix "unused invalid_chunk" with STBI_FAILURE_USERMSG
 - stb#1404: stb_image: Fix gif two_back memory address
 - stb#1420: stb_image: Improve error reporting if file operations fail within *_from_file functions
 - stb#1445: stb_vorbis: Few static analyzers fixes
 - stb#1487: stb_vorbis: Fix residue classdata bounding for f->temp_memory_required
 - stb#1490: stb_vorbis: Fix broken clamp in codebook_decode_deinterleave_repeat
 - stb#1496: stb_image: Fix pnm only build
 - stb#1497: stb_image: Fix memory leaks if stbi__convert failed
 - stb#1498: stb_vorbis: Fix memory leaks in stb_vorbis
 - stb#1499: stb_vorbis: Minor change to prevent the undefined behavior - left shift of a negative value
 - stb#1500: stb_vorbis: Fix signed integer overflow

Includes additional small fixes that I felt didn't warrant a separate PR.

PR note: the attached image should display correctly when using `printimage.com` built from this branch; version from `master` spews errors and crashes.

<details>
<summary>
Image
</summary>

media_Ej4qfYzXYAA69S1

</details>

### CODE REVIEW

**Review by jart — APPROVED** [review]
Looks good to me. Thanks for your contribution.

I just need one process hurdle from you first before I can merge. Can
you please send an email to jtunney@gmail.com saying "I intend to assign
you the copyright to the changes I contribute to Cosmopolitan"? It only
applies to the changes you choose to contribute. It only needs to happen
once. If your employer owns everything you do, then you'll need to clear
it with them first. Please do that if you need to and get it on the
record. Thanks!

**Comment by mataha** [comment]
> Looks good to me. Thanks for your contribution.
> 
> I just need one process hurdle from you first before I can merge. Can you please send an email to [jtunney@gmail.com](mailto:jtunney@gmail.com) saying "I intend to assign you the copyright to the changes I contribute to Cosmopolitan"? It only applies to the changes you choose to contribute. It only needs to happen once. If your employer owns everything you do, then you'll need to clear it with them first. Please do that if you need to and get it on the record. Thanks!

Already done; did it land in spam?

File: third_party/stb/stb_truetype.c [hunk 1]
```diff
@@ -1,29 +1,20 @@
-/*-*- mode:c;indent-tabs-mode:nil;c-basic-offset:3;tab-width:8;coding:utf-8 -*-│
-│vi: set net ft=c ts=3 sts=3 sw=3 fenc=utf-8                                :vi│
+/*-*- mode:c;indent-tabs-mode:nil;c-basic-offset:2;tab-width:8;coding:utf-8 -*-│
+│vi: set net ft=c ts=2 sts=2 sw=2 fenc=utf-8                                :vi│
 ╞══════════════════════════════════════════════════════════════════════════════╡
+│ Copyright 2023 Justine Alexandra Roberts Tunney                              │
 │                                                                              │
-│  stb_truetype                                                                │
```
Comment (jart) [inline third_party/stb/stb_truetype.c:5]: Could you put the copyright of the original authors back? Even though this is a public domain library, it's a bad look for me to impose further restrictions on its use, even if they are as minor as the ISC license. My intent is for the STB code in this folder of the cosmopolitan repo to keep being public domain.

Comment (mataha) [reply third_party/stb/stb_truetype.c]: Sure. I've reverted my changes.

**Pull request merged**